### PR TITLE
Add is_inf() and is_finite() (Issues #4100 and #2497)

### PR DESCRIPTION
**Pull request by steven-johnson**
attn @benoitsteiner

### CODE REVIEW

**Comment by steven-johnson** [comment]
(These are not particularly urgent at all but I needed a nice end-of-day task...)

File: src/CodeGen_LLVM.cpp [hunk 1]
```diff
@@ -3156,6 +3156,37 @@ void CodeGen_LLVM::visit(const Call *op) {
...
+        llvm::FastMathFlags safe_flags;
+        safe_flags.clear();
+        builder->setFastMathFlags(safe_flags);
+        builder->setDefaultFPMathTag(strict_fp_math_md);
+
+        // isinf(e) -> (fabs(e) == infinity)
+        Expr e = op->args[0];
+        internal_assert(e.type().is_float());
+        Expr inf = e.type().max();
+        codegen(abs(e) == inf);
```
Comment (abadams) [inline src/CodeGen_LLVM.cpp]: would codegen(strict_float(abs(e) == inf)) be equivalent but shorter?

File: src/CodeGen_LLVM.cpp [hunk 2]
```diff
@@ -3156,6 +3156,37 @@ void CodeGen_LLVM::visit(const Call *op) {
...
+        llvm::FastMathFlags safe_flags;
+        safe_flags.clear();
+        builder->setFastMathFlags(safe_flags);
+        builder->setDefaultFPMathTag(strict_fp_math_md);
+
+        // isfinite(e) -> (fabs(e) != infinity && !isnan(e)) -> (fabs(e) != infinity && e == e)
+        Expr e = op->args[0];
+        internal_assert(e.type().is_float());
+        Expr inf = e.type().max();
+        codegen(abs(e) != inf && e == e);
```
Comment (abadams) [inline src/CodeGen_LLVM.cpp]: Same comment here.

File: src/CodeGen_LLVM.cpp (hunk 1 above)
Comment (steven-johnson) [reply src/CodeGen_LLVM.cpp]: That's a good question -- I was following the pattern used by is_nan here. Maybe. Though I think a "righter" answer is probably to follow this with a fix for https://github.com/halide/Halide/issues/4281 (which I filed this morning), as the use of these calls outside a strict_float context in the first place is likely an error on the coder's part anyway.

**Comment by benoitsteiner** [comment]
Thanks for adding this.

**Pull request merged**